PhpUnit/DefinitionHasArgumentConstraint.php

```diff
@@ -17,13 +21,30 @@ public function __construct($argumentIndex, $expectedValue, $checkExpectedValue
     {
         parent::__construct();
 
-        $this->argumentIndex = (integer)$argumentIndex;
+        if (!(is_string($argumentIndex) || (is_int($argumentIndex) && $argumentIndex >= 0))) {
+            throw new \InvalidArgumentException('Expected either a string or a positive integer for $argumentIndex.');
+        }
+
+        if (is_string($argumentIndex) && '$' !== $argumentIndex[0]) {
```

Comment:
Technically, you would have to make sure first, that`$argumentIndex` is not the empty string.

Reply:
You're right.